app/src/main/java/com/halim/adam/cashmaster/AddNewBudgetRatio.java

```diff
@@ -0,0 +1,35 @@
+package com.halim.adam.cashmaster;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.EditText;
+
+public class AddNewBudgetRatio extends Activity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_add_new_budget_ratio);
+    }
+
+    public void GetInput(View view){
+        EditText inputName = findViewById(R.id.inputName);
+        EditText inputPortion = findViewById(R.id.inputPortion);
+
+        DatabaseHelper dbHelper = new DatabaseHelper(this);
+        dbHelper.InsertBudgetRatio(inputName.getText().toString(), Float.valueOf(inputPortion.getText().toString()));
+
+        Intent intent = new Intent(this, ViewBudgets.class);
+        startActivity(intent);
+        finish();
+    }
+
+    public void MoveToViewBudgetActivity(View view){
+        Intent intent = new Intent(this, ViewBudgets.class);
+        startActivity(intent);
+        finish();
```

Comment:
Is this intended? Do you intend to make sure that this activity cannot be returned to from other activity?

If you don't want this activity to be kept in the backstack you can use 
```
android:noHistory="true"
```
in the manifest

https://stackoverflow.com/questions/11308198/start-new-activity-and-finish-current-one-in-android?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

If that's not what you meant, you can just delete the `finish()` and it will work okay